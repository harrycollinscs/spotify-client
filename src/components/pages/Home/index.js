import { connect } from 'react-redux';
import Home from '../../templates/Home';
import store from '../../../store';
import { updateLoggedInStatus } from '../../../store/actions';

const mapStateToProps = () => ({
  isLoggedIn: store.getState().isLoggedIn,
  user: store.getState().user,
});

const mapDispatchToProps = dispatch => ({
  updateLoggedInStatus:  isLoggedIn => dispatch(updateLoggedInStatus(isLoggedIn)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Home);
