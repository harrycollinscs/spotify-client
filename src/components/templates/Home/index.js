import React from "react";
import Page from "../../atoms/Page";
import Hero from "../../molecules/Hero";

const locale = {
  preTitle: 'Hey',
  title: username => `Hey, ${username}`,
  content: 'Welcome back!'
}
const Home = ({ user }) => {
  return (
    <Page>
      <Hero
        title={user ? locale.title(user.id) : locale.preTitle}
        content={locale.content}
        withBackButton={false}
      />
      {/* <Content>{user ? <p>Hey, {user.id}!</p> : <p>Welcome back!</p>}</Content> */}
    </Page>
  );
};

export default Home;
