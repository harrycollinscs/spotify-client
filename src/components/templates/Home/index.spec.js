import React from "react";
import Home from "./index";
import { shallow } from "enzyme";

describe("Templates/Home", () => {
  const props = {
    user: {
      id: 'TestUsername',
    }
  }

  const component = <Home {...props} />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
