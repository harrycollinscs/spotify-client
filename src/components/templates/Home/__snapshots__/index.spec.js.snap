// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Home Should match snapshot 1`] = `
<Page>
  <Hero
    content="Welcome back!"
    title="Hey, TestUsername"
    withBackButton={false}
  />
</Page>
`;
